tpke/src/combine.rs

```diff
@@ -89,6 +90,28 @@ pub fn share_combine_fast<E: PairingEngine>(
     E::product_of_pairings(&pairing_product)
 }
 
+pub fn checked_share_combine_fast<R: RngCore, E: PairingEngine>(
+    pub_contexts: &[PublicDecryptionContextFast<E>],
+    ciphertexts: &[Ciphertext<E>],
+    decryption_shares: &[DecryptionShareFast<E>],
+    prepared_key_shares: &[E::G2Prepared],
+    rng: &mut R,
+) -> Result<E::Fqk> {
+    let is_valid_shares = batch_verify_decryption_shares(
+        pub_contexts,
+        ciphertexts,
+        // TODO: Figure out why batch verification takes an array of decryption shares and ciphertexts
+        &[decryption_shares.to_vec()],
```

Comment:
TODO: We can simplify this code now that we only have one share per participant. We can remove the `alpha_*` from the code and use a simplified equation (for one ciphertext instead of many).